Collect labels for deployments and statefulsets.

alamaison opened the pull request:
It builds but is untested because `go test` just says `?   	k8s.io/kube-state-metrics	[no test files]`.

I'm not a go programmer, so if this is not the correct way to run tests please let me know. 

I'm not sure I understand the significance of the default labels.  Copying how the service collector defines them, I've added `namespace` and the type of object being labelled.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kube-state-metrics/175)
<!-- Reviewable:end -->

brancz commented:
Thanks a lot for your contribution!

You can just run `make test-unit` to execute the tests locally.

We'll also need to you or your organization to sign the CLA for us to be able to accept these changes.

alamaison commented:
Thanks. I've added tests and fixed the errors.  Except one.  The stateful set label collection works as expected, one entry for each statefulset.  However the identical setup for deployments only gives labels for the first deployment.  Perhaps you could have a look to see if I'm doing something obviously wrong?

brancz commented on collectors/deployment.go:
```diff
@@ -160,4 +180,7 @@ func (dc *deploymentCollector) collectDeployment(ch chan<- prometheus.Metric, d
 	}
+
+	labelKeys, labelValues := kubeLabelsToPrometheusLabels(d.Labels)
+	addGauge(deploymentLabelsDesc(labelKeys), 1, labelValues...)
```
Just move these two lines to be the first metrics to be added (line 164). The reason these metrics are not showing up is the guard clause on line 173-175.

alamaison commented:
Derp.  Thanks for the help.  I've updated with a fix.

brancz commented:
Thanks a lot for your contribution!
